composition/content_merge.go

```diff
@@ -55,12 +58,20 @@ func NewContentMerge(metaJSON map[string]interface{}) *ContentMerge {
 	return cntx
 }
 
+func (cntx *ContentMerge) SetDeduplicationStrategy(strategy StylesheetDeduplicationStrategy) {
+	cntx.stylesheetDeduplicationStrategy = strategy
+}
+
 func (cntx *ContentMerge) collectStylesheets(f Fragment) {
 	cntx.stylesheets = append(cntx.stylesheets, f.Stylesheets()...)
 }
 
 func (cntx *ContentMerge) writeStylesheets(w io.Writer) {
-	for _, attrs := range cntx.stylesheets {
+	stylesheets := cntx.stylesheets
```

Comment:
Yeah i see, but wouldn't it be more correct to also deduplicate the stylesheets in the context object? This way someone implementing something in the future could erroneously use cntx.stylesheets thinking it was already deduplicated. I do not see why they should not be deduplicated too.